### PR TITLE
refactor ode tests, add scipy benchmark

### PR DESCRIPTION
**Pull request by jacobjinkelly**
Adding on to #2818 and #2819

### CODE REVIEW

File: tests/ode_test.py [hunk 1]
```diff
@@ -31,77 +34,124 @@ def num_float_bits(dtype):
   return dtypes.finfo(dtypes.canonicalize_dtype(dtype)).bits
 
 
-class JetTest(jtu.JaxTestCase):
```
Comment (mattjj) [inline tests/ode_test.py:34]: Hah, you caught my copy-paste bug!

File: tests/ode_test.py [hunk 2]
```diff
@@ -31,77 +34,124 @@ def num_float_bits(dtype):
...
+class ODETest(jtu.JaxTestCase):
+
+  def scipy_benchmark(self, fun, y0, tspace, *args, tol=1e-1):
```
Comment (mattjj) [inline tests/ode_test.py:39]: Maybe rename to `scipy_check` or `check_against_scipy`?

**Review by mattjj — APPROVED** [review]
This is really nice work! It's dramatically improving our ODE tests to check correctness against scipy. (And I'm impressed at how much of the codebase you're familiar with now!)

Two minor comments: one a name change, and one a suggestion for keeping the performance benchmarking (rather than correctness checking) bit in ode.py. But both are easy to adjust in follow-ups if we want, so LGTM.

**Comment by mattjj** [comment]
Hmm, the test that fails on Travis passes foor me locally:

```
In [1]: run tests/ode_test.py ODETest.test_odeint_vmap_grad
Running tests under Python 3.7.4: /usr/local/google/home/mattjj/miniconda3/bin/python
[ RUN      ] ODETest.test_odeint_vmap_grad
[       OK ] ODETest.test_odeint_vmap_grad
----------------------------------------------------------------------
Ran 1 test in 3.932s

OK
```

Any idea what's up with that failure?

File: tests/ode_test.py (hunk 2 above)
Comment (jacobjinkelly) [reply tests/ode_test.py]: Good idea, makes it much clearer!

**Comment by jacobjinkelly** [comment]
Thanks for the kind words @mattjj, glad to contribute!

1. RE: performance benchmarking of `odeint`. Totally agree, will follow up and add this to `ode.py`, perhaps refactoring using [benchmark.py](https://github.com/google/jax/blob/master/benchmarks/benchmark.py#L38) (Although not sure how to use that and stay in `ode.py`, or if we should compare to performance of `scipy` as well)

2. `test_odeint_vmap_grad` also passes for me locally, so also not sure why Travis is breaking unfortunately.

**Comment by mattjj** [comment]
Hrm, I'll look into Travis...

**Comment by mattjj** [comment]
I was able to reproduce the error with `pytest tests -W ignore` (not running multi-process), but it's still a puzzle to me.

**Comment by mattjj** [comment]
Progress! No solutions yet. The un-parallelized tests take forever to run, so I keep paging this out to work on other stuff.

**Comment by jacobjinkelly** [comment]
Woohoo! Thanks for fixing this @mattjj!

**Comment by mattjj** [comment]
I'm not sure why this bug manifest in exactly this way, but it seems to be sorted!

**Pull request merged**